src/coreclr/tools/Common/TypeSystem/Common/Utilities/CustomAttributeTypeNameParser.cs

```diff
@@ -179,21 +179,18 @@ private TypeDesc GetSimpleTypeFromModule(TypeName typeName, ModuleDesc module)
                     TypeDesc type = GetSimpleTypeFromModule(typeName.DeclaringType, module);
                     if (type == null)
                         return null;
-                    return ((MetadataType)type).GetNestedType(TypeNameHelpers.Unescape(typeName.Name));
+                    return ((MetadataType)type).GetNestedType(TypeName.Unescape(typeName.Name));
                 }
 
-                string fullName = TypeNameHelpers.Unescape(typeName.FullName);
-
                 if (_canonGenericResolver != null)
                 {
+                    string fullName = TypeName.Unescape(typeName.FullName);
                     TypeDesc canonType = _canonGenericResolver(module, fullName);
                     if (canonType != null)
                         return canonType;
                 }
 
-                (string typeNamespace, string name) = TypeNameHelpers.Split(fullName);
-
-                return module.GetType(typeNamespace, name, throwIfNotFound: false);
```

Comment:
The tests are failing with:

```
Type Library.Not\\+Nested' should have been kept in assembly EscapedTypeNames
Type Library.Not\+Nested+Nes\\ted' should have been kept in assembly EscapedTypeNames
Type `Library.Not\+Nested+Nes/ted' should have been kept in assembly EscapedTypeNames
```

Could there be a problem with escaping of the original `(string typeNamespace, string name) = TypeNameHelpers.Split(fullName);` and the new `typeName.Namespace` and `typeName.Name`?